### PR TITLE
Sockets-Bita

### PR DESCRIPTION
**Pull request by Bitaman**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | How to write methods. Knowing ruby I only needed to use new syntaxes but the logig of writing method was the same and it was much easier doing this project this time.
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | It was a bit more confusing finding information about javascript maybe I am not familiar enough but I think it is hard to understand javascript code. 
What was a challenge you were able to overcome on this assignment? | trying the forEach loop a couple of times and getting problems with that lead me to find the for loop " for(const letter of word)" which I liked so much and used everywhere. I like this for loop a lot more than forEach.
What has been interesting and positive about learning a new programming language? |The fact that you learn much faster when you know one language is so exciting. Although I am a bit concern that I am going to end up mixing both languages a lot. I had to look up ruby syntaxes a couple of times doing white boarding and it's only the first week of javascript!!!  
What is something to focus on your learnings in JavaScript in the next week? | I am trying to understand jquery as much as I can as I think that was the most confusing topic of the week.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,8 +1,149 @@
...
+    for (const letter in letters) {
+      while (letters[letter] > 0) {
+        alphabets.push(letter);
+        letters[letter] -= 1;
+      }
+    }
+    const letterHand = [];
+    let limit = 0;
+    while (limit < 10) {
+      let index = Math.floor((Math.random() * 100) + 1);
```
Comment (dHelmgren) [inline src/adagrams.js:40]: This function won't be able to produce 0, so your math is off somewhat.

File: src/adagrams.js [hunk 2]
```diff
@@ -1,8 +1,149 @@
...
+      Z: 10,
+
+    };
+    let playWord = word.toUpperCase().split('');
+    let total = 0;
+    if (playWord.length == 0) {
+      return total;
+    }
+    playWord.forEach(function (letter) {
+      total += scoreChart[letter];
```
Comment (dHelmgren) [inline src/adagrams.js:104]: Nice work with the hash!

**Comment by dHelmgren** [comment]
# JS Adagrams
## What We're Looking For


Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | Yes
Small commits with meaningful commit messages | you seem to have seen your own mistake. :P
**Code Requirements** |
`drawLetters` method | Good, see comment
Uses appropriate data structure to store the letter distribution | yes!
All tests for `drawLetters` pass | yes
`usesAvailableLetters` method | Yes
All tests for `usesAvailableLetters` pass | yes
`scoreWord` method | yes
Uses appropriate data structure to store the letter scores | yes
All tests for `scoreWord` pass | yes
`highestScoreFrom` method | yes
Appropriately handles edge cases for tie-breaking logic | yes
All tests for `highestScoreFrom` pass | yes
**Overall** | Great work on this project! In this project you’ve taken some interesting logic and worked it into JavaScript syntax. I’m adding a few comments on places that this could be better and minor suggestions for improvement, but overall, you’ve done a great job of practicing syntax.
